salt/_states/metalk8s_kubernetes.py

```diff
@@ -1,1938 +1,164 @@
-    redis-master-deployment:
-      kubernetes.deployment_present:
-        - name: redis-master
-        - source: salt://k8s/redis-master-deployment.yml
-      require:
-        - pip: kubernetes-python-module
-
-    # kubernetes service as specified inside of
-    # a file containing the definition of the the
-    # service using the official kubernetes format
-    redis-master-service:
-      kubernetes.service_present:
-        - name: redis-master
-        - source: salt://k8s/redis-master-service.yml
-      require:
-        - kubernetes.deployment_present: redis-master
-
-    # kubernetes deployment as specified inside of
-    # a file containing the definition of the the
-    # deployment using the official kubernetes format
-    # plus some jinja directives
-     nginx-source-template:
-      kubernetes.deployment_present:
-        - source: salt://k8s/nginx.yml.jinja
-        - template: jinja
-      require:
-        - pip: kubernetes-python-module
-
-
-    # Kubernetes secret
-    k8s-secret:
-      kubernetes.secret_present:
-        - name: top-secret
-          data:
-            key1: value1
-            key2: value2
-            key3: value3
-
-.. versionadded: 2017.7.0
-'''
-from __future__ import absolute_import
-
-import copy
-import logging
-import yaml
-
-# Import 3rd-party libs
-from salt.ext import six
-import salt.utils.dictdiffer
-
-log = logging.getLogger(__name__)
+.. todo::
 
+   - Add support for patching objects (needed for annotations/labels/taints)
+   - Document usage of these state functions
+   - Support using a source file instead of inlining the manifest, and also
+     support the common Salt templating arguments
+"""
+from salt.exceptions import CommandExecutionError
+from salt.utils import yaml
 
 __virtualname__ = 'metalk8s_kubernetes'
 
 
 def __virtual__():
-    '''
-    Only load if the kubernetes module is available in __salt__
-    '''
-    if 'metalk8s_kubernetes.ping' not in __salt__:
-        return False, '`metalk8s_kubernetes.ping` not available'
-    else:
-        return __virtualname__
+    if 'metalk8s_kubernetes.create_object' not in __salt__:
+        return False, 'Missing `metalk8s_kubernetes` execution module'
+    return __virtualname__
 
 
-def _error(ret, err_msg):
-    '''
-    Helper function to propagate errors to
-    the end user.
-    '''
-    ret['result'] = False
-    ret['comment'] = err_msg
-    return ret
-
-
-def deployment_absent(name, namespace='default', **kwargs):
-    '''
-    Ensures that the named deployment is absent from the given namespace.
-
-    name
-        The name of the deployment
+def object_absent(name, manifest=None, **kwargs):
+    """Ensure that the object is absent.
 
-    namespace
-        The name of the namespace
-    '''
+    Arguments:
+        name (str): Path to a manifest yaml file or just a name
+        manifest (dict): Manifest content
+    """
+    ret = {'name': name, 'changes': {}, 'result': True, 'comment': ''}
```

Comment:
We can but as we never return False I don't know if it's really usefull (the only way to get a false is because of an exception raised and salt automatically set the result to false)